login-service/src/strategies/oauth/oauth.service.ts

```diff
@@ -0,0 +1,156 @@
+import { Inject, Injectable } from "@nestjs/common";
+import { StrategyInstanceService } from "src/model/services/strategy-instance.service";
+import { StrategiesService } from "../../model/services/strategies.service";
+import * as passportOauth from "passport-oauth2";
+import { StrategyInstance } from "src/model/postgres/StrategyInstance.entity";
+import * as passport from "passport";
+import { LoginUserService } from "src/model/services/login-user.service";
+import { UserLoginDataService } from "src/model/services/user-login-data.service";
+import { AuthFunction, AuthResult, AuthStateData } from "../AuthResult";
+import { StrategyUsingPassport } from "../StrategyUsingPassport";
+import { JwtService } from "@nestjs/jwt";
+import { UserLoginData } from "src/model/postgres/UserLoginData.entity";
+import { ActiveLoginService } from "src/model/services/active-login.service";
+import { checkType } from "../utils";
+
+@Injectable()
+export class OauthStrategyService extends StrategyUsingPassport {
```

Comment:
If I understand the code correctly, this uses GitHub for Oauth 
maybe the name is a bit misleading?

Reply:
Refactured in d5f4daa